components/eventing-controller/pkg/backend/jetstreamv2/jetstream_internal_unit_test.go

```diff
@@ -0,0 +1,425 @@
+package jetstreamv2
+
+import (
+	"sync"
+	"testing"
+
+	backenderrors "github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/jetstreamv2/errors"
+	"github.com/stretchr/testify/assert"
+
+	kymalogger "github.com/kyma-project/kyma/common/logging/logger"
+	"github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha2"
+	"github.com/kyma-project/kyma/components/eventing-controller/logger"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/cleaner"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/metrics"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/mocks"
+	subtesting "github.com/kyma-project/kyma/components/eventing-controller/testing/v2"
+	"github.com/nats-io/nats.go"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_SyncNATSConsumers_ForEmptyTypes(t *testing.T) {
+	// given
+	callback := func(m *nats.Msg) {}
+	subWithOneType := subtesting.NewSubscriptionWithEmptyTypes()
+
+	// when
+	js := JetStream{}
+	err := js.syncNATSConsumers(subWithOneType, callback)
+
+	// then
+	assert.NoError(t, err)
+}
+
+// Test_SyncNATSConsumers tests creation/binding of the consumer and NATS Subscriptions.
+func Test_SyncNATSConsumers(t *testing.T) {
+	testEnv := getTestEnvironmentWithMock(t)
+	callback := func(m *nats.Msg) {}
+	subWithOneType := subtesting.NewSubscriptionWithOneType()
+	_, jsSubKey, defaultConsumerInfo, _ := generateConsumerInfra(testEnv.jsBackend, subWithOneType, subWithOneType.Status.Types[0])
+	subscriberMock := &mocks.Subscriber{}
```

Comment:
It would be easier (I think) to understand for the test reader if we can simply construct these objects manually. Maybe we can refactor this as we did with `Test_CheckNATSSubscriptionsCount` 🤔